ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -302,33 +322,55 @@ export const PageProviderDisplayBehavior = [
     },
 
     deselectItem(item) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllChecked) {
         this.$.list.deselectItem(item);
         this._updateFlags();
       }
     },
 
     deselectIndex(index) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllChecked) {
         this.$.list.deselectIndex(index);
         this._updateFlags();
       }
     },
 
     selectAll() {
-      this.items.forEach(
-        function(item) {
-          this.selectItem(item);
-        }.bind(this.$.list),
-      );
-      this._updateFlags();
+      if (this.selectionEnabled && this.selectAllEnabled) {
+        this._isSelectAllChecked = true;
+        // select the visible first (the others can be deferred)
+        const start = Math.max(0, this.$.list.firstVisibleIndex - 15);
+        const end = Math.min(this.items.length, this.$.list.lastVisibleIndex + 15);
```

Comment:
I'm a bit concerned about the magic number 15, which might not be enough when we have smaller items (i.e. zoomed in with enlarged height of the `nuxeo-results`), but I did not test it. What about coming up with an heuristic? Fetching `N` amount of items before and after the current range, where `N` is the number of visible items on the list (i.e. `const N = this.$.list._physicalItems.length;`)?

Reply:
🤔 It could be. Miguel manifested the same concern about the magic number 15. I actually calculated it based on our default `pageSize` and from my empiric testing. But I'm going to try with your suggestion. Thanks